projects/plugins/jetpack/extensions/shared/register-jetpack-block.native.js

```diff
@@ -14,43 +19,87 @@ function requiresPlan( unavailableReason, details ) {
 /**
  * Registers a gutenberg block if the availability requirements are met.
  *
- * @param {string} name - The block's name.
+ * @param {string} nameOrMetadata - The block's name or metadata object. Jetpack blocks must be
+ * registered with a name prefixed with `jetpack/`. This function accepts an unprefixed name too,
+ * though (it'd handle both `business-hours` and `jetpack/business-hours` similarly, for instance).
  * @param {object} settings - The block's settings.
  * @param {object} childBlocks - The block's child blocks.
+ * @param {boolean} prefix - Should this block be prefixed with `jetpack/`?
  * @returns {object|boolean} Either false if the block is not available, or the results of `registerBlockType`
  */
-export default function registerJetpackBlock( name, settings, childBlocks = [] ) {
-	const { available, details, unavailableReason } = getJetpackExtensionAvailability( name );
+export default function registerJetpackBlock(
+	nameOrMetadata,
+	settings,
+	childBlocks = [],
+	prefix = true
+) {
+	const name = typeof nameOrMetadata === 'string' ? nameOrMetadata : nameOrMetadata.name;
+	const isNamePrefixed = name.startsWith( JETPACK_PREFIX );
+	const rawName = isNamePrefixed ? name.slice( JETPACK_PREFIX.length ) : name;
+
+	const { available, details, unavailableReason } = getJetpackExtensionAvailability( rawName );
 
 	const requiredPlan = requiresPlan( unavailableReason, details );
+	const jpPrefix = prefix || isNamePrefixed ? JETPACK_PREFIX : '';
 
 	if ( ! available && ! requiredPlan ) {
 		if ( 'production' !== process.env.NODE_ENV ) {
 			// eslint-disable-next-line no-console
 			console.warn(
-				`Block ${ name } couldn't be registered because it is unavailable (${ unavailableReason }).`
+				`Block ${ rawName } couldn't be registered because it is unavailable (${ unavailableReason }).`
 			);
 		}
 		return false;
 	}
 
-	const result = registerBlockType( `jetpack/${ name }`, {
+	const prefixedName = jpPrefix + rawName;
+	const result = registerBlockType( nameOrMetadata === 'object' ? nameOrMetadata : prefixedName, {
 		...settings,
-		title: betaExtensions.includes( name ) ? `${ settings.title } (beta)` : settings.title,
+		title: betaExtensions.includes( prefixedName ) ? `${ settings.title } (beta)` : settings.title,
```

Comment:
**Discrepancy between native and web version:** In the web version, the beta suffix in the title is added using WP hooks. Eventually, we adopt this approach to remove this line.

https://github.com/Automattic/jetpack/blob/eebad7d58da476499ccc090d3b78c02547918c33/projects/plugins/jetpack/extensions/shared/register-jetpack-block.js#L49-L52

https://github.com/Automattic/jetpack/blob/3178c7f15114d90cc91f31332eef9cf060735f1d/projects/plugins/jetpack/extensions/editor.js#L81-L106

Reply:
Thanks Carlos, I note you've mentioned this could be a future improvement. Do you think there's anything we need to change here in the context of this PR? Or should we track this somewhere as a future improvement?

Reply:
> [...] Do you think there's anything we need to change here in the context of this PR? Or should we track this somewhere as a future improvement?

My comment was merely a heads-up that this changed in the web version.  So I think there's nothing to apply in this PR related to this. As you pointed out, we could consider tracking it in a new GitHub issue. I'll create one as a follow-up.

Reply:
**To close the loop:** I created https://github.com/wordpress-mobile/gutenberg-mobile/issues/6357 as a follow-up